src/check/parse/Parser.zig

```diff
@@ -117,7 +117,9 @@ pub fn pushDiagnostic(self: *Parser, tag: IR.Diagnostic.Tag, region: IR.Region)
 /// add a malformed token
 pub fn pushMalformed(self: *Parser, comptime t: type, tag: IR.Diagnostic.Tag) t {
     const pos = self.pos;
-    self.advanceOne(); // TODO: find a better point to advance to
+    if (self.peek() != .EndOfFile) {
```

Comment:
I want pushMalformed to include a advance_to Token.Tag so that it takes care of this (it will also need a start position in order to create a real Region)

Reply:
Sounds good. 👍